### PR TITLE
feat(Select): add hasClear property

### PR DESCRIPTION
**Pull request by Marginy605**

### CODE REVIEW

File: src/components/Select/types.ts [hunk 1]
```diff
@@ -65,6 +72,7 @@ export type SelectProps<T = any> = QAProps &
         multiple?: boolean;
         filterable?: boolean;
         disablePortal?: boolean;
+        hasClear?: boolean;
```
Comment (korvin89) [inline src/components/Select/types.ts:75]: Lets add this type to readme file

File: src/components/Select/components/SelectClear/SelectClear.scss [hunk 2]
```diff
@@ -0,0 +1,63 @@
+@use '../../../../../styles/mixins';
+@use '../../../variables';
+
+$block: '.#{variables.$ns-new}select-clear';
+$sHeight: 24px;
```
Comment (korvin89) [inline src/components/Select/components/SelectClear/SelectClear.scss:5]: U can get these variables [here](https://github.com/gravity-ui/uikit/blob/main/src/components/variables.scss)

File: src/components/Select/components/SelectClear/SelectClear.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    const {size, onClick, onMouseEnter, onMouseLeave, renderIcon} = props;
+    const icon = renderIcon ? (
+        renderIcon()
+    ) : (
+        <Icon className={selectClearBlock('clear')} data={CrossIcon} />
+    );
+    return (
+        <button
+            className={selectClearBlock({size})}
+            aria-label="Clear"
```
Comment (korvin89) [inline src/components/Select/components/SelectClear/SelectClear.tsx:20]: I think we should add text with i18n

File: src/components/Select/components/SelectControl/SelectControl.tsx [hunk 4]
```diff
@@ -45,12 +55,66 @@ export const SelectControl = React.forwardRef<HTMLElement, ControlProps>((props,
...
+        setIsDisabledButtonAnimation(true);
+    }, []);
+    const enableButtonAnimation = React.useCallback(() => {
+        setIsDisabledButtonAnimation(false);
+    }, []);
+    const handleOnClearIconClick = React.useCallback(
+        (e: React.MouseEvent) => {
+            // return animation on clear click
+            setIsDisabledButtonAnimation(false);
+            clearValue?.(e);
```
Comment (korvin89) [inline src/components/Select/components/SelectControl/SelectControl.tsx:98]: Why clearValue is optional? It is our inner method and we always add them to the component

**Review by korvin89 — COMMENTED** [review]
Could u add some test for new property?

File: src/components/Select/components/SelectControl/SelectControl.tsx [hunk 5]
```diff
@@ -59,38 +123,48 @@ export const SelectControl = React.forwardRef<HTMLElement, ControlProps>((props,
...
-                data-qa={qa}
-                name={name}
-                disabled={disabled}
-                onClick={toggleOpen}
-                onKeyDown={onKeyDown}
+            <div
+                className={selectControlBlock(controlMods)}
+                ref={handleControlRef}
+                role="group"
+                tabIndex={-1}
```
Comment (korvin89) [inline src/components/Select/components/SelectControl/SelectControl.tsx:138]: Why `tabIndex={-1}`?

File: src/components/Select/Select.tsx [hunk 6]
```diff
@@ -134,6 +135,14 @@ export const Select = React.forwardRef<HTMLButtonElement, SelectProps>(function
         [handleSelection, multiple],
     );
 
+    const clearValue = React.useCallback(
+        (e?: React.MouseEvent | React.KeyboardEvent<HTMLElement>) => {
+            e?.stopPropagation();
```
Comment (korvin89) [inline src/components/Select/Select.tsx:140]: I think we do not need that anymore

File: src/components/Select/types.ts (hunk 1 above)
Comment (Marginy605) [reply src/components/Select/types.ts:75]: fixed

File: src/components/Select/components/SelectClear/SelectClear.scss (hunk 2 above)
Comment (Marginy605) [reply src/components/Select/components/SelectClear/SelectClear.scss]: fixed

File: src/components/Select/components/SelectClear/SelectClear.tsx (hunk 3 above)
Comment (Marginy605) [reply src/components/Select/components/SelectClear/SelectClear.tsx]: fixed

File: src/components/Select/components/SelectControl/SelectControl.tsx (hunk 5 above)
Comment (Marginy605) [reply src/components/Select/components/SelectControl/SelectControl.tsx]: fixed

File: src/components/Select/Select.tsx (hunk 6 above)
Comment (Marginy605) [reply src/components/Select/Select.tsx]: fixed

**Comment by Marginy605** [comment]
> Could u add some test for new property?

added tests

File: src/components/Select/components/SelectClear/SelectClear.tsx [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+            tabIndex={0}
```
[the remark this answers is not in the document]
Comment (Marginy605) [reply src/components/Select/components/SelectClear/SelectClear.tsx]: fixed

File: src/components/Select/__tests__/Select.clear.test.tsx [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+    jest.clearAllMocks();
+});
+
+const onUpdate = jest.fn();
+
+describe('Select clear', () => {
+    test.each<[string, Partial<SelectProps>]>([
+        ['single', {hasClear: true, multiple: false}],
+        ['multiple', {hasClear: true, multiple: true}],
+    ])('initial', async () => {
```
[the remark this answers is not in the document]
Comment (Marginy605) [reply src/components/Select/__tests__/Select.clear.test.tsx:20]: fixed

File: src/components/Select/components/SelectClear/SelectClear.tsx [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+    const icon = renderIcon ? (
+        renderIcon()
+    ) : (
+        <Icon className={selectClearBlock('clear')} data={CrossIcon} />
+    );
+    return (
+        <button
+            className={selectClearBlock({size})}
+            aria-label={i18n('label_clear')}
+            /*tabIndex={0}*/
```
[the remark this answers is not in the document]
Comment (Marginy605) [reply src/components/Select/components/SelectClear/SelectClear.tsx:22]: fixed

**Review by korvin89 — APPROVED** [review]

**Pull request merged**